R/bainCommon.R

```diff
@@ -1107,200 +1173,178 @@
+      #   ggplot2::scale_color_manual(values = rep("black",200), guide=ggplot2::guide_legend(nrow=10)) +
+      ggplot2::scale_y_continuous(name = options[["dependent"]], breaks = yBreaks, limits = range(yBreaks)) +
+      ggplot2::scale_x_continuous(name = options[["fixedFactors"]], breaks = 1:length(varLevels), labels = varLevels) +
+      jaspGraphs::geom_rangeframe() +
+      jaspGraphs::themeJaspRaw()
     plot$plotObject <- p
   }
 }
 
 .plotModelProbabilitiesTTests <- function(x, type) {
-  
   if (type == 1 || type == 2 || type == 3) {
-    labs <- c(gettext("H0"), gettext("H1"))
+    labels <- c(gettext("H0"), gettext("H1"))
   } else if (type == 4) {
-    labs <- c(gettext("H1"), gettext("H2"))
+    labels <- c(gettext("H1"), gettext("H2"))
   } else if (type == 5) {
-    labs <- c(gettext("H0"), gettext("H1"), gettext("H2"))
+    labels <- c(gettext("H0"), gettext("H1"), gettext("H2"))
   }
-  labels <- rev(labs)
-  
+
   if (type == 1) {
-    values <- x$fit$PMPb
+    postProb <- x$fit$PMPb
   } else {
-    values <- na.omit(x$fit$PMPa)
+    postProb <- na.omit(x$fit$PMPa)
   }
-  
-  ggdata <- data.frame(lab = labs, PMP = values)
-  
-  p <- ggplot2::ggplot(data = ggdata, mapping = ggplot2::aes(x = "", y = PMP, fill = lab)) +
+
+  plotData <- data.frame(x = labels, y = postProb)
+  yBreaks <- cumsum(rev(postProb)) - rev(postProb) / 2
+  p <- ggplot2::ggplot(data = plotData, mapping = ggplot2::aes(x = "", y = y, fill = x)) +
     ggplot2::geom_bar(stat = "identity", width = 1e10, color = "black", size = 1) +
-    ggplot2::geom_col() +
     ggplot2::coord_polar(theta = "y", direction = -1) +
-    ggplot2::labs(x = "", y = "") +
-    ggplot2::theme(panel.grid = ggplot2::element_blank(), legend.position = "none") +
-    ggplot2::scale_y_continuous(breaks = cumsum(rev(values)) - rev(values)/2, labels = labels) +
-    ggplot2::theme(panel.background = ggplot2::element_blank(),
-                   axis.text=ggplot2::element_text(size=17, color = "black"),
-                   plot.title = ggplot2::element_text(size=18, hjust = .5),
-                   axis.ticks.y = ggplot2::element_blank()) +
-    ggplot2::scale_fill_brewer(palette="Set1")
-  
+    ggplot2::scale_x_discrete(name = NULL) +
+    ggplot2::scale_y_continuous(name = NULL, breaks = yBreaks, labels = rev(plotData[["x"]])) +
+    ggplot2::scale_fill_brewer(palette = "Set1") +
+    jaspGraphs::themeJaspRaw(legend.position = "none") +
+    ggplot2::theme(axis.ticks.y = ggplot2::element_blank())
+
   return(p)
 }
 
 .plotModelProbabilitiesRegression <- function(x) {
-  
-  PMPa <- na.omit(x$fit$PMPa)
-  PMPb <- x$fit$PMPb
-  numH <- length(PMPa)
-  P_lables <- paste(gettext("H"), 1:numH, sep = "")
-  ggdata1 <- data.frame(lab = P_lables, PMP = PMPa)
-  ggdata2 <- data.frame(lab = c(P_lables, gettext("Hu")), PMP = PMPb)
-  
+  postProbA <- na.omit(x$fit$PMPa)
+  postProbB <- x$fit$PMPb
```

Comment:
```suggestion
  postProbA <- na.omit(x$fit$PMPa)
  postProbB <- x$fit$PMPb
```

why not `na.omit(x$fit$PMPb)` as well? Can it have NAs too or that a nonissue?

Reply:
`x$fit$PMPa` always has 1 `NA` value in it, while `x$fit$PMPb` does not.